### ISSUE TITLE
Comments aren't "universally" supported

### ISSUE THREAD

**Issue by mrjbq7** [factor/factor#737]
For example, these don't work:

```factor
USING: sequences ! asdfsadf
math ;
```

```factor
STRUCT: foo
    { a int } ! this is a comment
    { b int } ;
```

**Comment by mrjbq7** [comment]
See also #1243

**Comment by mrjbq7** [comment]
I tried this patch and it seems to give bootstrap problems with urls ebnf:

```
resource:basis/help/markup/markup.factor

3: USING: accessors arrays assocs classes colors colors.constants
9: splitting strings urls vectors vocabs vocabs.loader words
                         ^
resource:basis/urls/urls.factor

44: EBNF: parse-url
66: ;EBNF
         ^
T{ not-compiled f ( gensym ) ~unbalanced-branches-error~ }

The following restarts are available:

:1      Load resource:basis/urls/urls.factor again
:2      Load resource:basis/help/markup/markup.factor again
:3      Load resource:basis/help/crossref/crossref.factor again
4 :res  Load resource:basis/help/help.factor again
Factor 0.98 x86.64 (1707, heads/master-399d01f56e, Tue Aug  4 18:14:07 2015)
[Clang (GCC 4.2.1 Compatible Apple LLVM 6.1.0 (clang-602.0.53))] on macosx
:errors - show 1 compiler errors
IN: scratchpad ^C^C^C
Interrupted
Starting low level debugger...
Basic commands:
  q ^D             -- quit Factor
  c                -- continue executing Factor - NOT SAFE
  t                -- throw exception in Factor - NOT SAFE
  .s .r .c         -- print data, retain, call stacks
  help             -- full help, including advanced commands

> ^C
```

**Comment by erg** [comment]
```factor
"CHAR: !" eval( -- obj )
```

The parser has to understand whether the next token is to be interpreted as a raw string or not.

**Comment by mrjbq7** [comment]
Damnit, thats right, parse-token is used for "raw".

**Comment by mrjbq7** [comment]
Fixed in 52a3f6f3097f7760dfd72d234bf856b27bf6abac and 77c256412dcb6924545e8c1c0d01a195d10f6ea8.
